tests/v2/validation/nodescaling/replace.go

```diff
@@ -64,20 +64,25 @@ func MatchNodeToRole(t *testing.T, client *rancher.Client, clusterID string, isE
 func ReplaceNodes(t *testing.T, client *rancher.Client, clusterName string, isEtcd bool, isControlPlane bool, isWorker bool) {
 	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
 	require.NoError(t, err)
-	numOfNodesBeforeDeletion, nodesToDelete := MatchNodeToRole(t, client, clusterID, isEtcd, isControlPlane, isWorker)
+	numOfNodesBeforeDeletion, nodeToDelete := MatchNodeToRole(t, client, clusterID, isEtcd, isControlPlane, isWorker)
 
-	for i := range nodesToDelete {
-		machineToDelete, err := client.Steve.SteveType(machineSteveResourceType).ByID("fleet-default/" + nodesToDelete[i].Annotations[machineSteveAnnotation])
+	for i := range nodeToDelete {
+		machineToDelete, err := client.Steve.SteveType(machineSteveResourceType).ByID("fleet-default/" + nodeToDelete[i].Annotations[machineSteveAnnotation])
 		require.NoError(t, err)
 
-		logrus.Infof("Replacing node: " + nodesToDelete[i].NodeName)
+		logrus.Infof("Deleting node: " + nodeToDelete[i].NodeName)
 		err = client.Steve.SteveType(machineSteveResourceType).Delete(machineToDelete)
 		require.NoError(t, err)
 
+		err = nodestat.IsNodeDeleted(client, nodeToDelete[i].NodeName, clusterID)
+		require.NoError(t, err)
+
+		logrus.Infof("Replacing node: " + nodeToDelete[i].NodeName)
+
 		err = clusters.WaitClusterToBeUpgraded(client, clusterID)
 		require.NoError(t, err)
 
-		logrus.Infof("Checking if node %s is replaced", nodesToDelete[i].NodeName)
+		logrus.Infof("Checking if node %s is replaced", nodeToDelete[i].NodeName)
 		_, err = nodestat.IsNodeReplaced(client, machineToDelete.ID, clusterID, numOfNodesBeforeDeletion)
```

Comment:
this function is technically just doing a check if the node is deleted. Could you check if this function is used anywhere else, and if not just remove it? Its redundant with the addition of `isNodeDeleted`

Reply:
It's being used in the auto replace test function below.

Reply:
Resolved offline